### PR TITLE
Edit dialog

### LINKED ISSUE #18: update readme

### LINKED ISSUE #71: Use dialog to edit notes

### LINKED ISSUE #73: Refresh button
Add a refresh button in the app to force the new data to be fetched (for a collaborative scenario).

### PR DESCRIPTION
**Pull request by swouf**
Notes can now be edited with a dialog. Also, I added a refresh button to manually sync with the backend.

close #71 
close #73 
close #18

### CODE REVIEW

File: src/components/main/note/EditableText.tsx [hunk 1]
```diff
@@ -1,71 +1,50 @@
...
-        // The class content is styled in `note_style.css`
-        className="content"
-        // eslint-disable-next-line react/no-danger
-        dangerouslySetInnerHTML={{ __html: content.innerHTML }}
-      />
-    );
-  },
-);
+const EditableText = (props: EditableTextProps): JSX.Element => {
+  const { text } = props;
```
Comment (spaenleh) [inline src/components/main/note/EditableText.tsx:12]: Can you use props de-structuring in the argument instead ?

File: src/components/main/NoteContainer.tsx [hunk 2]
```diff
@@ -96,14 +96,14 @@ const NoteContainer = (props: NoteContainerInterface): JSX.Element => {
   };
 
   const handleCanvasClick = (event: React.MouseEvent): void => {
-    if (!noteBeingEditedId && !noteBeingTransformedId) {
+    if (!noteBeingEditedId && !noteBeingTransformedId && event.detail === 2) {
```
[the remark this answers is not in the document]
Comment (swouf) [reply src/components/main/NoteContainer.tsx:99]: It checks the event is a double click. I put a comment.

**Comment by swouf** [comment]
Here is the screenshot of the dialog. I also updated the README.
screenshot_dialog_SN_builder

File: src/components/main/Canvas.tsx [hunk 3]
```diff
@@ -39,6 +42,8 @@ const MainContainer = styled('div')(() => ({
 const Canvas = (): JSX.Element => {
   const [backgroundToggleSetting, setBackgroundToggleSetting] = useState(false);
   const context = useContext(Context);
```
Comment (spaenleh) [inline src/components/main/Canvas.tsx:44]: I think you can remove this lien since you are using the `useLocalContext` hook which is the same.

**Review by spaenleh — APPROVED** [review]
Looks good ! Nice work ! 📽️

File: src/components/main/Canvas.tsx (hunk 3 above)
Comment (swouf) [reply src/components/main/Canvas.tsx:44]: Thanks, I'll fix this.

**Pull request merged**